error: `cuda.h` file not found

```
FAILED: caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_broadcast.cu.o 
cd /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math && /usr/bin/cmake -E make_directory /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/./torch_hip_generated_broadcast.cu.o -P /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_broadcast.cu.o.cmake
In file included from /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/caffe2/utils/math/broadcast.cu:3:
In file included from /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/caffe2/core/context_gpu.h:8:
/home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/caffe2/core/common_gpu.h:5:10: fatal error: 'cuda.h' file not found
#include <cuda.h>
         ^~~~~~~~
1 error generated when compiling for host.
CMake Error at torch_hip_generated_broadcast.cu.o.cmake:138 (message):
  Error generating
  /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/./torch_hip_generated_broadcast.cu.o


[3767/4523] cd /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math && /usr/bin/cmake -E make_directory /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/./torch_hip_generated_transpose.cu.o -P /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_transpose.cu.o.cmake
FAILED: caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_transpose.cu.o 
cd /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math && /usr/bin/cmake -E make_directory /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/./torch_hip_generated_transpose.cu.o -P /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_transpose.cu.o.cmake
In file included from /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/caffe2/utils/math/transpose.cu:7:
/home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/caffe2/core/common_gpu.h:5:10: fatal error: 'cuda.h' file not found
#include <cuda.h>
         ^~~~~~~~
1 error generated when compiling for host.
CMake Error at torch_hip_generated_transpose.cu.o.cmake:138 (message):
  Error generating
  /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/./torch_hip_generated_transpose.cu.o


[3768/4523] cd /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip && /usr/bin/cmake -E make_directory /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/./torch_hip_generated_assert_op.hip.o -P /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/torch_hip_generated_assert_op.hip.o.cmake
[3769/4523] cd /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip && /usr/bin/cmake -E make_directory /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/./torch_hip_generated_roi_align_rotated_op.hip.o -P /home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/torch_hip_generated_roi_align_rotated_op.hip.o.cmake
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "setup.py", line 732, in <module>
    build_deps()
  File "setup.py", line 311, in build_deps
    build_caffe2(version=version,
  File "/home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/tools/build_pytorch_libs.py", line 62, in build_caffe2
    cmake.build(my_env)
  File "/home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/tools/setup_helpers/cmake.py", line 345, in build
    self.run(build_args, my_env)
  File "/home/acxz/vcs/git/github/rocm-arch/python-pytorch-rocm/src/pytorch-1.6.0rc6-rocm/tools/setup_helpers/cmake.py", line 141, in run
    check_call(command, cwd=self.build_dir, env=env)
  File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--target', 'install', '--config', 'Release', '--', '-j', '4']' returned non-zero exit status 1.
==> ERROR: A failure occurred in build().
```

upstream issue: https://github.com/pytorch/pytorch/issues/42430

Well, I remembered it wrong, turns out it's this error for python-pytorch-rocm for me as well.
```
FAILED: caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_broadcast.cu.o 
cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/./torch_hip_generated_broadcast.cu.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_broadcast.cu.o.cmake
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/utils/math/broadcast.cu:3:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/core/context_gpu.h:8:
/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/core/common_gpu.h:5:10: fatal error: 'cuda.h' file not found
#include <cuda.h>
         ^~~~~~~~
1 error generated when compiling for host.
CMake Error at torch_hip_generated_broadcast.cu.o.cmake:138 (message):
  Error generating
  /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/./torch_hip_generated_broadcast.cu.o


[3767/4523] cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/core && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/core/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/core/./torch_hip_generated_context_gpu.cu.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/core/torch_hip_generated_context_gpu.cu.o.cmake
FAILED: caffe2/CMakeFiles/torch_hip.dir/core/torch_hip_generated_context_gpu.cu.o 
cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/core && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/core/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/core/./torch_hip_generated_context_gpu.cu.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/core/torch_hip_generated_context_gpu.cu.o.cmake
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/core/context_gpu.cu:8:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/c10/cuda/CUDACachingAllocator.h:4:
/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/c10/cuda/CUDAStream.h:6:10: fatal error: 'cuda_runtime_api.h' file not found
#include <cuda_runtime_api.h>
         ^~~~~~~~~~~~~~~~~~~~
1 error generated when compiling for host.
CMake Error at torch_hip_generated_context_gpu.cu.o.cmake:138 (message):
  Error generating
  /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/core/./torch_hip_generated_context_gpu.cu.o


[3768/4523] cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/image && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/image/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/image/./torch_hip_generated_transform_gpu.cu.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/image/torch_hip_generated_transform_gpu.cu.o.cmake
FAILED: caffe2/CMakeFiles/torch_hip.dir/image/torch_hip_generated_transform_gpu.cu.o 
cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/image && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/image/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/image/./torch_hip_generated_transform_gpu.cu.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/image/torch_hip_generated_transform_gpu.cu.o.cmake
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/image/transform_gpu.cu:1:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/core/context_gpu.h:8:
/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/core/common_gpu.h:5:10: fatal error: 'cuda.h' file not found
#include <cuda.h>
         ^~~~~~~~
1 error generated when compiling for host.
CMake Error at torch_hip_generated_transform_gpu.cu.o.cmake:138 (message):
  Error generating
  /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/image/./torch_hip_generated_transform_gpu.cu.o


[3769/4523] cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/./torch_hip_generated_transpose.cu.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_transpose.cu.o.cmake
FAILED: caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_transpose.cu.o 
cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/./torch_hip_generated_transpose.cu.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_transpose.cu.o.cmake
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/utils/math/transpose.cu:7:
/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/core/common_gpu.h:5:10: fatal error: 'cuda.h' file not found
#include <cuda.h>
         ^~~~~~~~
1 error generated when compiling for host.
CMake Error at torch_hip_generated_transpose.cu.o.cmake:138 (message):
  Error generating
  /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/./torch_hip_generated_transpose.cu.o


[3770/4523] cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/./torch_hip_generated_reduce.cu.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_reduce.cu.o.cmake
FAILED: caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_reduce.cu.o 
cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/./torch_hip_generated_reduce.cu.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_reduce.cu.o.cmake
/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/utils/math/reduce.cu:9:10: fatal error: 'cub/block/block_reduce.cuh' file not found
#include <cub/block/block_reduce.cuh>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated when compiling for host.
CMake Error at torch_hip_generated_reduce.cu.o.cmake:138 (message):
  Error generating
  /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/./torch_hip_generated_reduce.cu.o


[3771/4523] cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/./torch_hip_generated_elementwise.cu.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_elementwise.cu.o.cmake
FAILED: caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_elementwise.cu.o 
cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/./torch_hip_generated_elementwise.cu.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_elementwise.cu.o.cmake
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/utils/math/elementwise.cu:10:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/core/context_gpu.h:8:
/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/core/common_gpu.h:5:10: fatal error: 'cuda.h' file not found
#include <cuda.h>
         ^~~~~~~~
1 error generated when compiling for host.
CMake Error at torch_hip_generated_elementwise.cu.o.cmake:138 (message):
  Error generating
  /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/./torch_hip_generated_elementwise.cu.o


[3772/4523] cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/./torch_hip_generated_math_gpu.cu.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/torch_hip_generated_math_gpu.cu.o.cmake
FAILED: caffe2/CMakeFiles/torch_hip.dir/utils/torch_hip_generated_math_gpu.cu.o 
cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/./torch_hip_generated_math_gpu.cu.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/torch_hip_generated_math_gpu.cu.o.cmake
/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/utils/math_gpu.cu:10:10: fatal error: 'cub/block/block_reduce.cuh' file not found
#include <cub/block/block_reduce.cuh>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated when compiling for host.
CMake Error at torch_hip_generated_math_gpu.cu.o.cmake:138 (message):
  Error generating
  /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/./torch_hip_generated_math_gpu.cu.o


[3773/4523] cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/./torch_hip_generated_accuracy_op.hip.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/torch_hip_generated_accuracy_op.hip.o.cmake
[3774/4523] cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/./torch_hip_generated_group_norm_op.hip.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/torch_hip_generated_group_norm_op.hip.o.cmake
[3775/4523] cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/./torch_hip_generated_batch_permutation_op.hip.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/torch_hip_generated_batch_permutation_op.hip.o.cmake
[3776/4523] cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/./torch_hip_generated_acos_op.hip.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/torch_hip_generated_acos_op.hip.o.cmake
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/hip/acos_op.hip:2:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/acos_op.h:6:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/elementwise_ops.h:15:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/utils/eigen_utils.h:6:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/cmake/../third_party/eigen/Eigen/Core:202:
/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/cmake/../third_party/eigen/Eigen/src/Core/arch/GPU/PacketMathHalf.h:149:1: warning: non-void function does not return a value [-Wreturn-type]
}
^
1 warning generated when compiling for gfx803.
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/hip/acos_op.hip:2:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/acos_op.h:6:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/elementwise_ops.h:15:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/utils/eigen_utils.h:6:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/cmake/../third_party/eigen/Eigen/Core:202:
/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/cmake/../third_party/eigen/Eigen/src/Core/arch/GPU/PacketMathHalf.h:149:1: warning: non-void function does not return a value [-Wreturn-type]
}
^
1 warning generated when compiling for gfx900.
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/hip/acos_op.hip:2:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/acos_op.h:6:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/elementwise_ops.h:15:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/utils/eigen_utils.h:6:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/cmake/../third_party/eigen/Eigen/Core:202:
/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/cmake/../third_party/eigen/Eigen/src/Core/arch/GPU/PacketMathHalf.h:149:1: warning: non-void function does not return a value [-Wreturn-type]
}
^
1 warning generated when compiling for gfx906.
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/hip/acos_op.hip:2:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/acos_op.h:6:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/elementwise_ops.h:15:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/utils/eigen_utils.h:6:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/cmake/../third_party/eigen/Eigen/Core:202:
/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/cmake/../third_party/eigen/Eigen/src/Core/arch/GPU/PacketMathHalf.h:149:1: warning: non-void function does not return a value [-Wreturn-type]
}
^
1 warning generated when compiling for gfx908.
[3777/4523] cd /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip && /usr/bin/cmake -E make_directory /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/./torch_hip_generated_abs_op.hip.o -P /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/operators/hip/torch_hip_generated_abs_op.hip.o.cmake
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/hip/abs_op.hip:2:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/abs_op.h:6:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/elementwise_ops.h:15:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/utils/eigen_utils.h:6:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/cmake/../third_party/eigen/Eigen/Core:202:
/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/cmake/../third_party/eigen/Eigen/src/Core/arch/GPU/PacketMathHalf.h:149:1: warning: non-void function does not return a value [-Wreturn-type]
}
^
1 warning generated when compiling for gfx803.
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/hip/abs_op.hip:2:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/abs_op.h:6:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/elementwise_ops.h:15:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/utils/eigen_utils.h:6:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/cmake/../third_party/eigen/Eigen/Core:202:
/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/cmake/../third_party/eigen/Eigen/src/Core/arch/GPU/PacketMathHalf.h:149:1: warning: non-void function does not return a value [-Wreturn-type]
}
^
1 warning generated when compiling for gfx900.
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/hip/abs_op.hip:2:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/abs_op.h:6:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/elementwise_ops.h:15:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/utils/eigen_utils.h:6:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/cmake/../third_party/eigen/Eigen/Core:202:
/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/cmake/../third_party/eigen/Eigen/src/Core/arch/GPU/PacketMathHalf.h:149:1: warning: non-void function does not return a value [-Wreturn-type]
}
^
1 warning generated when compiling for gfx906.
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/hip/abs_op.hip:2:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/abs_op.h:6:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/operators/elementwise_ops.h:15:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/utils/eigen_utils.h:6:
In file included from /home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/cmake/../third_party/eigen/Eigen/Core:202:
/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/cmake/../third_party/eigen/Eigen/src/Core/arch/GPU/PacketMathHalf.h:149:1: warning: non-void function does not return a value [-Wreturn-type]
}
^
1 warning generated when compiling for gfx908.
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "setup.py", line 732, in <module>
    build_deps()
  File "setup.py", line 311, in build_deps
    build_caffe2(version=version,
  File "/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/tools/build_pytorch_libs.py", line 62, in build_caffe2
    cmake.build(my_env)
  File "/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/tools/setup_helpers/cmake.py", line 345, in build
    self.run(build_args, my_env)
  File "/home/USERNAME/python-pytorch-rocm/src/pytorch-1.6.0-rocm/tools/setup_helpers/cmake.py", line 141, in run
    check_call(command, cwd=self.build_dir, env=env)
  File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--target', 'install', '--config', 'Release', '--', '-j', '12']' returned non-zero exit status 1.
```

Building `1.6.0-4`:
<details>
<summary>At first I was like…</summary>
[22/763] Building HIPCC object caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_elementwise.cu.o
FAILED: caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_elementwise.cu.o 
cd /home/USER/Build/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math && /usr/bin/cmake -E make_directory /home/USER/Build/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/. && /usr/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=RELEASE -D generated_file:STRING=/home/USER/Build/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/./torch_hip_generated_elementwise.cu.o -P /home/USER/Build/python-pytorch-rocm/src/pytorch-1.6.0-rocm/build/caffe2/CMakeFiles/torch_hip.dir/utils/math/torch_hip_generated_elementwise.cu.o.cmake
In file included from /home/USER/Build/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/utils/math/elementwise.cu:10:
In file included from /home/USER/Build/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/core/context_gpu.h:8:
/home/USER/Build/python-pytorch-rocm/src/pytorch-1.6.0-rocm/caffe2/core/common_gpu.h:5:10: fatal error: 'cuda.h' file not found
#include <cuda.h>
</details>

<details>
<summary>When trying again in a clean builddir and waiting for hours…</summary>
==> Finished building: python-pytorch-rocm 1.6.0-4 (Do 06 Aug 2020 02:36:15 CEST)
</details>

<details>
<summary>Unfortunately…</summary>
python-pytorch-rocm: /usr/lib/libuv.so exists in file system (owned by libuv)
python-pytorch-rocm: /usr/lib/libuv.so.1 exists in file system (owned by libuv)
python-pytorch-rocm: /usr/lib/libuv.so.1.0.0 exists in file system (owned by libuv)
</details>

I am overwriting this for now to test, but it might need sorting out.

<details>
<summary>Because also…</summary>
(1/1) Installing python-pytorch-rocm              [#####################################################] 100%
ldconfig: /usr/lib/libuv.so.1 is not a symlink
</details>

<details>
<summary>But finally…</summary>
In [5]: import torch
In [6]: torch.cuda.is_available()
Out[6]: True
In [7]: torch.cuda.current_device()
Out[7]: 0
In [8]: torch.cuda.device(0)
Out[8]: <torch.cuda.device at 0x7fbe913d9670>
In [9]: torch.cuda.device_count()
Out[9]: 1
In [10]: torch.cuda.get_device_name()
Out[10]: 'Vega 10 XL/XT [Radeon RX Vega 56/64]'
In [11]: vega56 = torch.device("cuda")
In [12]: torch.rand(10).to(vega56)
Out[12]: 
tensor([4.0125e-01, 6.2919e-01, 7.3994e-01, 2.5187e-01, 8.3741e-01, 5.2937e-01,
        9.2326e-01, 9.6917e-05, 5.2456e-01, 4.1349e-01], device='cuda:0')
</details>

Works as intended in this small check. Cheers man!

--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = python-pytorch-rocm
 	pkgdesc = Tensors and Dynamic neural networks in Python with strong GPU acceleration
 	pkgver = 1.7.1
-	pkgrel = 3
+	pkgrel = 4
 	url = https://pytorch.org
 	arch = x86_64
 	license = BSD
@@ -40,6 +40,7 @@ pkgbase = python-pytorch-rocm
 	source = nccl_version.patch
 	source = disable_non_x86_64.patch
 	source = find-hsa-runtime.patch::https://patch-diff.githubusercontent.com/raw/pytorch/pytorch/pull/45550.patch
+	source = fix-hip-version.patch
 	sha256sums = SKIP
 	sha256sums = 83c81ec6a461110da6ae6182529f58100986b068c5182ca62cd53c648b4e4fb0
 	sha256sums = 26b1dd596f1e21a011ee18cab939924483d6c6d4d98e543bf76f5a9312d54d67
@@ -47,6 +48,7 @@ pkgbase = python-pytorch-rocm
 	sha256sums = e4a96887b41cbdfd4204ce5f16fcb16a23558d23126331794ab6aa30a66f2e0d
 	sha256sums = d3ef8491718ed7e814fe63e81df2f49862fffbea891d2babbcb464796a1bd680
 	sha256sums = SKIP
+	sha256sums = a972c80561320e36b6c32933294b2b6c19715233e15c7976f2eb2db44c436b3b
 
 pkgname = python-pytorch-rocm
 	pkgdesc = Tensors and Dynamic neural networks in Python with strong GPU acceleration (with ROCM)

--- a/PKGBUILD
+++ b/PKGBUILD
@@ -7,7 +7,7 @@ pkgname=("python-pytorch-rocm" "python-pytorch-opt-rocm")
 _pkgname="pytorch"
 pkgver=1.7.1
 _pkgver=1.7.1
-pkgrel=3
+pkgrel=4
 pkgdesc="Tensors and Dynamic neural networks in Python with strong GPU acceleration"
 arch=('x86_64')
 url="https://pytorch.org"
@@ -22,14 +22,16 @@ source=("${_pkgname}-${pkgver}::git+https://github.com/pytorch/pytorch.git#tag=v
         use-system-libuv2.patch
         nccl_version.patch
         disable_non_x86_64.patch
-        "find-hsa-runtime.patch::https://patch-diff.githubusercontent.com/raw/pytorch/pytorch/pull/45550.patch")
+        "find-hsa-runtime.patch::https://patch-diff.githubusercontent.com/raw/pytorch/pytorch/pull/45550.patch"
+        fix-hip-version.patch)
 sha256sums=('SKIP'
             '83c81ec6a461110da6ae6182529f58100986b068c5182ca62cd53c648b4e4fb0'
             '26b1dd596f1e21a011ee18cab939924483d6c6d4d98e543bf76f5a9312d54d67'
             '7b65c3b209fc39f92ba58a58be6d3da40799f1922910b1171ccd9209eda1f9eb'
             'e4a96887b41cbdfd4204ce5f16fcb16a23558d23126331794ab6aa30a66f2e0d'
             'd3ef8491718ed7e814fe63e81df2f49862fffbea891d2babbcb464796a1bd680'
-            'SKIP')
+            'SKIP'
+            'a972c80561320e36b6c32933294b2b6c19715233e15c7976f2eb2db44c436b3b')
 
 prepare() {
   cd "${_pkgname}-${pkgver}"
@@ -52,6 +54,9 @@ prepare() {
 
   # https://github.com/pytorch/pytorch/pull/45550
   patch -Np1 -i "${srcdir}"/find-hsa-runtime.patch
+  
+  # patch newer HIP version patch define
+  patch -Np1 -i "${srcdir}"/fix-hip-version.patch
 
   # remove local nccl
   rm -rf third_party/nccl/nccl
@@ -67,13 +72,13 @@ prepare() {
 
   # Check tools/setup_helpers/cmake.py, setup.py and CMakeLists.txt for a list of flags that can be set via env vars.
   export USE_MKLDNN=ON
-  export BUILD_CUSTOM_PROTOBUF=OFF
+  export BUILD_CUSTOM_PROTOBUF=ON
   # export BUILD_SHARED_LIBS=OFF
   export USE_FFMPEG=ON
   export USE_GFLAGS=ON
   export USE_GLOG=ON
   export BUILD_BINARY=ON
-  # export USE_OPENCV=ON
+  export USE_OPENCV=ON
   export USE_SYSTEM_NCCL=ON
   # export USE_SYSTEM_LIBS=ON
   export NCCL_VERSION=$(pkg-config nccl --modversion)

--- a/fix-hip-version.patch
+++ b/fix-hip-version.patch
@@ -1,0 +1,21 @@
+diff --git a/third_party/eigen/Eigen/src/Core/util/ConfigureVectorization.h b/third_party/eigen/Eigen/src/Core/util/ConfigureVectorization.h
+index 68765d4b2..df2d46832 100644
+--- a/third_party/eigen/Eigen/src/Core/util/ConfigureVectorization.h
++++ b/third_party/eigen/Eigen/src/Core/util/ConfigureVectorization.h
+@@ -401,11 +401,11 @@
+   #include <hip/hip_fp16.h>
+ 
+   #define HIP_PATCH_WITH_NEW_FP16 18215
+-  #if (HIP_VERSION_PATCH < HIP_PATCH_WITH_NEW_FP16)
+-    #define EIGEN_HAS_OLD_HIP_FP16
+-    // Old HIP implementation does not have a explicit typedef for "half2"
+-    typedef __half2 half2;
+-  #endif
++//  #if (HIP_VERSION_PATCH < HIP_PATCH_WITH_NEW_FP16)
++//    #define EIGEN_HAS_OLD_HIP_FP16
++//    // Old HIP implementation does not have a explicit typedef for "half2"
++//    typedef __half2 half2;
++//  #endif
+ 
+ #endif
+

